File: front-end/src/app/service-models/app/v1/license/license-type-s-m.ts
import { CoinManagementServiceModelBase } from '../../base/coin-management-service-model-base';
import { RoleTypeSM } from '../../enums/role-type-s-m.enum';
import { PermissionSM } from '../client/permission-s-m';

export class LicenseTypeSM extends CoinManagementServiceModelBase<number> {
    title!: string;
    description!: string;
    validityInDays!: number;
    amount!: number;
    licenseTypeCode!: string;
    stripePriceId!: string;
    isPredefined!: boolean;
    validFor!: RoleTypeSM;
    permissionIds!: Array<number>;
    permissions!: Array<PermissionSM>;
}

File: backend/payroll-backend/Components/Siffrum.Web.Payroll.ServiceModels/v1/License/LicenseTypeSM.cs
using Siffrum.Web.Payroll.ServiceModels.Base;
using Siffrum.Web.Payroll.ServiceModels.Enums;
using Siffrum.Web.Payroll.ServiceModels.v1.Client;
using System.Collections.Generic;

namespace Siffrum.Web.Payroll.ServiceModels.v1.License
{
    public class LicenseTypeSM : SiffrumPayrollServiceModelBase<int>
    {
        public string Title { get; set; }
        public string? Description { get; set; }
        public int ValidityInDays { get; set; }
        public double Amount { get; set; }
        public string LicenseTypeCode { get; set; }
        public string StripePriceId { get; set; }
        public bool IsPredefined { get; set; }
        public RoleTypeSM ValidFor { get; set; }
        public List<int>? PermissionIds { get; set; }
        public List<PermissionSM>? Permissions { get; set; }
    }
}
